hop/models.py

```diff
@@ -138,6 +139,54 @@ def load_file(cls, filename):
             return cls.load(f)
 
 
+@dataclass
+class GCNTextNotice(MessageModel):
+    """A GCN Notice in the plain text format.
+
+    The original message data is stored in the raw property,
+    and the parsed message is available via the fields property,
+    which is a dictionary.
+    """
+
+    raw: bytes
+    fields: dict
+
+    def serialize(self):
+        return {"format": format_name(type(self)), "content": self.raw}
+
+    @classmethod
+    def deserialize(cls, data):
+        fields = {}
+        line_pattern = re.compile("([^:]*): *(.*)")
```

Comment:
I considered it cleaner because it can handle a more complex structure if needed, and one can see at a glance what it expects the data to look like. It's also handling a bit of the whitespace (and could deal with more of it, if desired, e.g. to eliminate the need for calling `strip` in various places). Ultimately, it doesn't make much difference, though.